server/aesdsocket.c

```diff
@@ -0,0 +1,335 @@
+
+{
+
+     openlog(NULL, 0, LOG_USER);
+
+// Register signal_handler as our signal handler for SIGINT
+if (signal (SIGINT, signal_handler) == SIG_ERR) 
+{
+   syslog(LOG_ERR, "Cannot handle SIGINT!\n");
+   graceful_exit();
+   exit (EXIT_FAILURE);
+}
+
+//Register signal_handler as our signal handler for SIGTERM.
+
+if (signal (SIGTERM, signal_handler) == SIG_ERR) 
+{
+ syslog(LOG_ERR, "Cannot handle SIGTERM!\n");
+  graceful_exit();
+ exit (EXIT_FAILURE);
+
+}
+
+
+    bool createDeamon = false;
+   
+    if (argc == 2) 
+    {
+        if (!strcmp(argv[1], "-d")) 
+        {
+            createDeamon = true;
+        } 
+
+        else 
+        {
+           syslog(LOG_ERR, "incorrect argument:%s", argv[1]);
+            return (-1);
+        }
+    }
+   
+
+    struct addrinfo hints;
+
+    memset(&hints, 0, sizeof(hints));
+
+    
+    hints.ai_family = AF_INET;
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_flags = AI_PASSIVE;
+    int res = getaddrinfo(NULL, (PORT), &hints, &updated_addr);
+    
+    if (res != 0) 
+    {
+        syslog(LOG_ERR, "Error:getaddrinfo() %s\n", gai_strerror(res));
+        graceful_exit();
+        return -1;
+    }
+
+
+    sfd = socket(updated_addr->ai_family, SOCK_STREAM, 0);
+    if (sfd < 0) 
+    {
+        syslog(LOG_ERR, "ERROR : Socket creation failed:%d\n", errno);
+        graceful_exit();
+        return -1;
+    }
+
+    if (setsockopt(sfd, SOL_SOCKET, SO_REUSEADDR, &(int){1}, sizeof(int)) < 0) 
+    {
+        syslog(LOG_ERR, "ERROR: setsockopt failed with errono:%d\n", errno);
+        graceful_exit();
+        return -1;
+    }
+
+
+    if (bind(sfd, updated_addr->ai_addr, updated_addr->ai_addrlen) < 0) 
+    {
+       syslog(LOG_ERR, "ERROR : failed to bind:%d\n", errno);;
+        graceful_exit();
+        return -1;
+    }
+
+
+   
+    if (listen(sfd, 20)) 
+    {
+       syslog(LOG_ERR, "ERROR:failed to listen %d\n", errno);
+       graceful_exit();
+        return -1;
+    }
+
+  
+
+    if (createDeamon == true) 
+    {
+         int ret_val = daemon(0,0);
+
+         if(ret_val<0)
+         {
+            syslog(LOG_ERR,"ERROR: Failed to create daemon");
+            graceful_exit();
+            return -1;
+
+         }    
+    }
+
+
+    while(true) 
+    {
+
+        struct sockaddr_in clientAddr;
+        socklen_t clientAddrSize = sizeof(clientAddr);
+
+        cfd = accept(sfd, (struct sockaddr*)&clientAddr, &clientAddrSize);
+        if(cfd < 0)
+        {
+            syslog(LOG_ERR, "ERROR: failed to accept: %s", strerror(errno));
+            graceful_exit();
+            return -1;
+        }
+
+        char client_ip[INET_ADDRSTRLEN];
+
+        inet_ntop(AF_INET, &(clientAddr.sin_addr), client_ip, INET_ADDRSTRLEN);
+
+        syslog(LOG_INFO, "connection successfully accepted%s \n\r",client_ip);
+        
+        
+        char recv_buf[2000];
+
+        while(true)
+        {
+            int readByteCounts = read(cfd, recv_buf, (2000));
```

Comment:
@SuryawanshiAbhishek what if previous msg is still there in the buffer or some garbage in which '\n' is present , would this code still work?